test/bdd/bluegreen_test.go

```diff
@@ -0,0 +1,193 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/cucumber/godog"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+
+	"github.com/argoproj/argo-rollouts/pkg/kubectl-argo-rollouts/cmd/promote"
+	"github.com/argoproj/argo-rollouts/pkg/kubectl-argo-rollouts/cmd/set"
+	"github.com/argoproj/argo-rollouts/pkg/kubectl-argo-rollouts/options"
+	argoexec "github.com/argoproj/pkg/exec"
+)
+
+type FunctionalTestContext struct {
+	RolloutName string
+	FilePath    string
+	Streams     genericclioptions.IOStreams
+	Options     *options.ArgoRolloutsOptions
+}
+
+func (c *FunctionalTestContext) iApplyManifest(fileName string) error {
+	folder := "../../examples"
+	filePath := folder + "/" + fileName
+	c.FilePath = filePath
+	_, err := argoexec.RunCommand("kubectl", argoexec.CmdOpts{}, "apply", "-f", filePath)
+	if err != nil {
+		return err
+	}
+
+	numAttempts := 4
+	isSuccess, err := retry(numAttempts, func() (bool, error) {
+		rolloutExists, err := argoexec.RunCommand("kubectl", argoexec.CmdOpts{}, "get", "rollout", c.RolloutName)
+		return rolloutExists != "", err
+	})
+	if err != nil {
+		return err
+	}
+	if !isSuccess {
+		return fmt.Errorf("Rollout not created")
```

Comment:
When this error (or any other error) returns, is it obvious which test is failing? If not, then this error won't make it easier for the developer to troubleshoot the issue.